src/css/style.css

```diff
@@ -173,18 +173,54 @@ line {
 }
 
 .modal {
+  display: none;
+  position: fixed;
+  z-index: 1;
+  left: 0;
+  top: 0;
+  width: 75%;
+  height: 100%;
+  overflow: auto;
   padding: 8px;
-  background-color: #ddd;
+  background-color: rgba(0,0,0,0.5);
   border: 1px solid black;
   border-radius: 6px;
   margin-bottom: 5px;
 }
 
 .modal p {
-  margin: 2px;
+  margin: 10px;
 }
 
 .error {
   color: red;
   padding-left: 5px;
 }
+
+.modal-content{
```

Comment:
To avoid confusion with non-legend modals, I would rename this `.legend-modal-content`.